laws/src/main/scala/cats/laws/MonadDeferLaws.scala

```diff
@@ -0,0 +1,62 @@
+package cats
+package laws
+
+import cats.syntax.all._
+import cats.laws.MonadDeferLaws.StatefulBox
+
+trait MonadDeferLaws[F[_]] extends ApplicativeEvalLaws[F] with MonadLaws[F] {
+  implicit override def F: MonadDefer[F]
+
+  def evalSequenceConsistentWithPureMap[A, B](a: A, b: B, f: (A, B) => A): IsEq[F[A]] = {
+    def tr(s: StatefulBox[A]) = s.transform(a => f(a, b))
+    val state1 = new StatefulBox(a)
+    val state2 = new StatefulBox(a)
+
+    val lh = F.delay(tr(state1))
+    val rh = F.pure(state2).map(tr)
+
+    lh.flatMap(_ => lh) <-> rh.flatMap(_ => rh)
```

Comment:
maybe it helps to add negative tests to be clear exactly what we are ruling out.